hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java

```diff
@@ -635,4 +641,24 @@ void setWriteBufferSize(int bufferSize) {
   void setEnableFlush(boolean enableFlush) {
     this.enableFlush = enableFlush;
   }
+
+  private String getMsiAuthEndpoint() throws IOException {
+    String authEndpoint = getPasswordString(
+        FS_AZURE_ACCOUNT_OAUTH_MSI_ENDPOINT);
+    if (Utils.isEmpty(authEndpoint)) {
+      authEndpoint =
+          AuthConfigurations.DEFAULT_FS_AZURE_ACCOUNT_OAUTH_MSI_ENDPOINT;
+    }
+    return authEndpoint;
+  }
+
+  private String getRefreshTokenAuthEndpoint() throws IOException {
+    String authEndpoint = getPasswordString(
+        FS_AZURE_ACCOUNT_OAUTH_REFRESH_TOKEN_ENDPOINT);
+    if (Utils.isEmpty(authEndpoint)) {
+      authEndpoint =
+          AuthConfigurations.DEFAULT_FS_AZURE_ACCOUNT_OAUTH_REFRESH_TOKEN_ENDPOINT;
+    }
+    return authEndpoint;
```

Comment:
given the empty check is on the trimmed value, should the authEndpoint be returned trimmed? as currenty a path like " http://something " with a space at either end will go through here and fail later, possibly with a meaningless message

Reply:
Currently the values are hard coded without space